dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -1081,6 +1083,36 @@ def test_subscript_dim_check():
     check_pipeline(data, single_op_pipeline, operator_fn=fn.subscript_dim_check, num_subscripts=1)
 
 
+def test_crop_argument_from_external_source():
+    """
+    Tests, if the fn.crop operator works correctly, when its actual batch size is lower
+    than max batch size.
+    """
+
+    @pipeline_def(batch_size=32, num_threads=4, device_id=0)
+    def pipeline():
+        images = fn.external_source(device="cpu", name="IMAGE", no_copy=False)
+        crop_x = fn.external_source(device="cpu", name="CROP_X", no_copy=False)
+        images = fn.decoders.image(images, device="mixed")
+
+        images = fn.crop(images, crop_pos_x=crop_x, crop_pos_y=0.05, crop_w=113, crop_h=149)
+        return images
+
+    pipe = pipeline()
+    pipe.build()
+
+    image_data = np.fromfile(
+        os.path.join(test_utils.get_dali_extra_path(), "db", "single", "jpeg", "100",
+                     "swan-3584559_640.jpg"),
+        dtype=np.uint8)
+    pipe.feed_input("IMAGE", [image_data])
+    pipe.feed_input("CROP_X", [np.float32(0.5)])
+    pipe.feed_input("IMAGE", [image_data])
+    pipe.feed_input("CROP_X", [np.float32(0.4)])
```

Comment:
Wow, I didn't see feed_input in quite some time.
I assume you checked that this test triggers the problem without the fix?

Reply:
Yes, without the fix wild segfault appears ;)